sentry-rails/spec/sentry/rails/tracing/active_record_subscriber_spec.rb

```diff
@@ -64,6 +64,38 @@
       expect(data["db.name"]).to include("db")
       expect(data["db.system"]).to eq("sqlite3")
     end
+
+    it "records database configuration if connection.pool is not available" do
+      # Some adapters (like CockroachDB) don't provide ConnectionPool,
+      # so we have to grab the configuration off the Connection itself.
+      # See https://github.com/getsentry/sentry-ruby/issues/2110
+      config = { adapter: "sqlite3", database: "dummy-database" }
+      allow_any_instance_of(ActiveRecord::ConnectionAdapters::SQLite3Adapter).to receive(:pool).and_return(nil)
```

Comment:
IMO, the fact that we need to artificially stub `pool` here means:

- We should bring in the adapter gem as a development dependency and write an isolated test just for it.
- Or, the adapter doesn't follow other adapters' convention to properly set things up.